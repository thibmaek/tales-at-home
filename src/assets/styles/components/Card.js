import { StyleSheet } from 'react-native';
import {
  globalStyle,
  white, mediumGray, lightBlack,
  fontIsSubHeader, 
} from 'src/assets/styles/vars';

export default StyleSheet.create({
  container: {
    width: 280,
    position: `relative`,
    marginLeft: 30,
    backgroundColor: white,
    borderRadius: 10,
  },
  innerContainer: {
    position: `absolute`, top: 0, zIndex: 3,
    borderRadius: 10, borderWidth: 10, borderColor: white, borderBottomWidth: 0,
    shadowColor: white, shadowOpacity: 1, shadowOffset: { width: 0, height: 20 }, shadowRadius: 7,
  },
  title: {
    ...globalStyle,
    ...fontIsSubHeader,
    fontSize: 14,
    paddingLeft: 20, paddingRight: 20, paddingTop: 20,
    borderRadius: 10,
  },
  meta: {
    ...globalStyle,
    fontSize: 11,
    paddingLeft: 20, paddingRight: 20, paddingBottom: 3,
    marginTop: 5,
    color: mediumGray,
  },
  contentScrollView: {
    marginRight: 10, marginTop: 60,
    paddingTop: 23,
  },
  content: {
    ...globalStyle,
    paddingLeft: 30, paddingRight: 20,
    fontSize: 14,
    lineHeight: 18,
    color: lightBlack,
  },
  buttonText: {
    ...globalStyle,
    fontWeight: `bold`,
  },
  button: {
    borderWidth: 0,
    shadowColor: white, shadowOpacity: 1, shadowRadius: 10,
    shadowOffset: { width: 0, height: - 25 },
    backgroundColor: white,
  },
});
