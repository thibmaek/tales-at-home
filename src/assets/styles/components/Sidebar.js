import { StyleSheet } from 'react-native';
import { globalStyle, accentBlue, white, shadowColorGray } from 'src/assets/styles/vars';

export default StyleSheet.create({
  sidebar: {
    width: 276,
    zIndex: 99,
    backgroundColor: white,
    shadowColor: shadowColorGray,
    shadowOffset: { width: 2.5, height: 5 },
    shadowOpacity: .5,
  },
  buttonBg: {
    height: 50,
    marginBottom: 0,
    backgroundColor: accentBlue,
    borderRadius: 0, borderColor: 0,
  },
  buttonText: {
    ...globalStyle,
    color: white,
    textAlign: `center`,
  },
});
