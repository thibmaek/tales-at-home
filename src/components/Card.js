import React, { PropTypes } from 'react';
import { View, Text, ScrollView } from 'react-native';

import Button from 'apsl-react-native-button';

import s from 'src/assets/styles/components/Card';

const Card = ({ title, meta, content, action }) => {
  return (
    <View style={s.container}>
      <View style={s.innerContainer}>
        <Text style={s.title}>{ title }</Text>
        <Text style={s.meta}>{ meta }</Text>
      </View>
      <ScrollView style={s.contentScrollView}>
        <Text style={s.content} selectable={true}>{ content }</Text>
      </ScrollView>
      { action ? <Button style={s.button} textStyle={s.buttonText}>{ action }</Button> : null }
    </View>
  );
};

Card.propTypes = {
  title: PropTypes.string.isRequired,
  meta: PropTypes.string,
  content: PropTypes.string,
  action: PropTypes.string,
};

export default Card;
