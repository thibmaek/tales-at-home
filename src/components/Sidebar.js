import React, { Component, PropTypes } from 'react';
import { View, ListView } from 'react-native';

import FamilyItem from 'src/components/FamilyItem';

import { sidebar } from 'src/assets/styles/components/Sidebar';

export default class Sidebar extends Component {
  constructor(props, context) {
    super(props, context);
    this.ds = new ListView.DataSource({ rowHasChanged: (r1, r2) => r1 !== r2 });
  }

  render() {
    return (
      <View style={sidebar}>
        <ListView
          dataSource={this.ds.cloneWithRows(this.props.families)}
          renderRow={family => <FamilyItem key={family.key} {...family} />}
        />
      </View>
    );
  }

  static propTypes = {
    families: PropTypes.array.isRequired,
  }
}
