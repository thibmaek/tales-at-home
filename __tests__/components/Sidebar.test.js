import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Sidebar from 'src/components/Sidebar';
import Card from 'src/components/Card';

jest.mock(`apsl-react-native-button`, () => `Button`);

describe(`Component: <Sidebar>`, () => {

  it(`should render without children passed`, () => {
    const tree = renderer.create(<Sidebar />).toJSON();
    expect(tree).toMatchSnapshot(`Sidebar`);
  });

  it(`should render wrapping, without children passed`, () => {
    const tree = renderer.create(<Sidebar></Sidebar>).toJSON();
    expect(tree).toMatchSnapshot(`Sidebar: Wrapping`);
  });

  it(`should render wrapping, with children passed`, () => {
    const tree = renderer.create(
      <Sidebar>
        <Card />
      </Sidebar>
    ).toJSON();
    expect(tree).toMatchSnapshot(`Sidebar: Wrapping children`);
  });

});

describe(`Component: <Sidebar {...props}>`, () => {

  it(`should render with an action prop`, () => {
    const tree = renderer.create(<Sidebar action={{
      type: `Neutral`,
      text: `A snapshot action`,
    }} />).toJSON();
    expect(tree).toMatchSnapshot(`Sidebar <Prop:title>`);
  });

});
