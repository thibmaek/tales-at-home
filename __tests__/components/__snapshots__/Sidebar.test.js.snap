exports[`Sidebar <Prop:title> 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 1)",
      "shadowColor": "rgb(188, 188, 188)",
      "shadowOffset": Object {
        "height": 5,
        "width": 2.5,
      },
      "shadowOpacity": 0.5,
      "width": 276,
      "zIndex": 8,
    }
  }>
  <DynamicButton
    action={undefined}
    style={
      Object {
        "backgroundColor": "rgba(0, 122, 255, 1)",
        "borderColor": 0,
        "borderRadius": 0,
        "height": 50,
      }
    }
    textStyle={
      Object {
        "color": "rgba(255, 255, 255, 1)",
        "fontFamily": "Bariol",
        "fontSize": 16,
        "fontWeight": "normal",
        "textAlign": "center",
      }
    }
    touchable="highlight">
    A snapshot action
  </DynamicButton>
</View>
`;

exports[`Sidebar 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 1)",
      "shadowColor": "rgb(188, 188, 188)",
      "shadowOffset": Object {
        "height": 5,
        "width": 2.5,
      },
      "shadowOpacity": 0.5,
      "width": 276,
      "zIndex": 8,
    }
  } />
`;

exports[`Sidebar: Wrapping 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 1)",
      "shadowColor": "rgb(188, 188, 188)",
      "shadowOffset": Object {
        "height": 5,
        "width": 2.5,
      },
      "shadowOpacity": 0.5,
      "width": 276,
      "zIndex": 8,
    }
  } />
`;

exports[`Sidebar: Wrapping children 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 1)",
      "shadowColor": "rgb(188, 188, 188)",
      "shadowOffset": Object {
        "height": 5,
        "width": 2.5,
      },
      "shadowOpacity": 0.5,
      "width": 276,
      "zIndex": 8,
    }
  }>
  <View
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 1)",
        "borderRadius": 10,
        "marginLeft": 30,
        "position": "relative",
        "width": 280,
      }
    }>
    <View
      style={
        Object {
          "borderBottomWidth": 0,
          "borderColor": "rgba(255, 255, 255, 1)",
          "borderRadius": 10,
          "borderWidth": 10,
          "position": "absolute",
          "shadowColor": "rgba(255, 255, 255, 1)",
          "shadowOffset": Object {
            "height": 20,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 7,
          "top": 0,
          "zIndex": 3,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "borderRadius": 10,
            "color": "rgba(74, 74, 74, 1)",
            "fontFamily": "Bariol",
            "fontSize": 14,
            "fontWeight": "bold",
            "paddingLeft": 20,
            "paddingRight": 20,
            "paddingTop": 20,
          }
        } />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(155, 155, 155, 1)",
            "fontFamily": "Bariol",
            "fontSize": 11,
            "fontWeight": "normal",
            "marginTop": 5,
            "paddingBottom": 3,
            "paddingLeft": 20,
            "paddingRight": 20,
          }
        } />
    </View>
    <ScrollView
      style={
        Object {
          "marginRight": 10,
          "marginTop": 60,
          "paddingTop": 23,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        selectable={true}
        style={
          Object {
            "color": "rgba(76, 76, 76, 1)",
            "fontFamily": "Bariol",
            "fontSize": 14,
            "fontWeight": "normal",
            "lineHeight": 18,
            "paddingLeft": 30,
            "paddingRight": 20,
          }
        } />
    </ScrollView>
  </View>
</View>
`;
